### PR TITLE
added caps swagger spec

### PR DESCRIPTION
**Pull request by junyilicaps**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by olydis** [comment]
@junyilicaps please remove the `swagger` folder from the path of your new file - see other Swaggers in this RP 🙂

**Review by olydis — CHANGES_REQUESTED** [review]
See other comment. Please also run this through a JSON formatter, there are some indentation inconsistencies.

File: specification/compute/resource-manager/Microsoft.Compute/2017-10-15-preview/swagger/computeArtifactPublishingService.json [hunk 1]
```diff
@@ -0,0 +1,870 @@
...
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
```
Comment (ravbhatnagar) [inline specification/compute/resource-manager/Microsoft.Compute/2017-10-15-preview/swagger/computeArtifactPublishingService.json]: Patch support missing for shared vm image.

File: specification/compute/resource-manager/Microsoft.Compute/2017-10-15-preview/swagger/computeArtifactPublishingService.json [hunk 2]
```diff
@@ -0,0 +1,870 @@
...
+            "description": "Accepted"
+          },
+          "204": {
+            "description": "If the shared VM image is deleted, this is an expected error code."
+          }
+        },
+        "x-ms-long-running-operation": true
+      }
+    },
+	 "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/sharedvmimages/{sharedVMImageName}/versions/{version}": {
```
Comment (ravbhatnagar) [inline specification/compute/resource-manager/Microsoft.Compute/2017-10-15-preview/swagger/computeArtifactPublishingService.json]: Is versions a proxy resource? If it is tracked, then PATCH support is missing. I believe this should not be tracked, but wanted to confirm

File: specification/compute/resource-manager/Microsoft.Compute/2017-10-15-preview/swagger/computeArtifactPublishingService.json [hunk 3]
```diff
@@ -0,0 +1,870 @@
+{
+  "swagger": "2.0",
```
Comment (ravbhatnagar) [inline specification/compute/resource-manager/Microsoft.Compute/2017-10-15-preview/swagger/computeArtifactPublishingService.json]: Not related to swagger - but please make sure these new options are added to the /operations API that compute owns.

File: specification/compute/resource-manager/Microsoft.Compute/2017-10-15-preview/swagger/computeArtifactPublishingService.json [hunk 4]
```diff
@@ -0,0 +1,870 @@
...
+    },
+	"SharedVMImageVersion": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/SharedVMImageVersionProperties"
+        }
+      },
+      "allOf": [{
+          "$ref": "#/definitions/Resource"
```
Comment (ravbhatnagar) [inline specification/compute/resource-manager/Microsoft.Compute/2017-10-15-preview/swagger/computeArtifactPublishingService.json]: If this is not a tracked resource, then it wont have location/tags properties

**Review by ravbhatnagar — COMMENTED** [review]
@junyilicaps please take a look at comments below

File: specification/compute/resource-manager/Microsoft.Compute/2017-10-15-preview/swagger/computeArtifactPublishingService.json (hunk 1 above)
Comment (junyilicaps) [reply specification/compute/resource-manager/Microsoft.Compute/2017-10-15-preview/swagger/computeArtifactPublishingService.json]: We don't have a patch API now.

File: specification/compute/resource-manager/Microsoft.Compute/2017-10-15-preview/swagger/computeArtifactPublishingService.json (hunk 4 above)
Comment (junyilicaps) [reply specification/compute/resource-manager/Microsoft.Compute/2017-10-15-preview/swagger/computeArtifactPublishingService.json]: Shared VM Image and Version are tracked resource.

File: specification/compute/resource-manager/Microsoft.Compute/2017-10-15-preview/computeArtifactPublishingService.json [hunk 5]
```diff
@@ -0,0 +1,870 @@
...
+        "showInGui": {
+          "type": "boolean",
+          "description": "Used by portal to show the image in dropdown."
+        },
+        "imageName": {
+          "type": "string",
+          "description": "ImageName for classic compute. If not specified we publish the image with Guid name and hide the image in listing in classic compute."
+        }
+      },
+      "required": [
```
Comment (olydis) [inline specification/compute/resource-manager/Microsoft.Compute/2017-10-15-preview/computeArtifactPublishingService.json]: empty `required` not valid according to spec, please remove (or make something required 😉 )

File: specification/compute/resource-manager/Microsoft.Compute/2017-10-15-preview/computeArtifactPublishingService.json [hunk 6]
```diff
@@ -0,0 +1,870 @@
...
+        "publisher": {
+          "type": "string",
+          "description": "Publisher name."
+        },
+        "product": {
+          "type": "string",
+          "description": "Product name."
+        }
+      },
+      "required": [
```
Comment (olydis) [inline specification/compute/resource-manager/Microsoft.Compute/2017-10-15-preview/computeArtifactPublishingService.json]: same as above

**Review by olydis — CHANGES_REQUESTED** [review]
Swagger schema violations

File: specification/compute/resource-manager/readme.md [hunk 7]
```diff
@@ -37,6 +37,14 @@ openapi-type: arm
 tag: package-2017-03
 ```
 
+### Tag: package-2017-10-15-preview
+
+These settings apply only when `--tag=package-2017-10-15-preview` is specified on the command line.
+
+``` yaml $(tag) == 'package-2017-10-15-preview'
+input-file:
+- Microsoft.Compute/2017-10-15-preview/computeArtifactPublishingService.json
```
Comment (olydis) [inline specification/compute/resource-manager/readme.md]: what you list here is what a "Compute SDK" would now contain - you probably wanna add
```
 - Microsoft.Compute/2017-03-30/compute.json
 - Microsoft.Compute/2017-03-30/disk.json
 - Microsoft.Compute/2017-03-30/runCommands.json
 - Microsoft.ContainerService/2017-01-31/containerService.json
```
to the list.

File: specification/compute/resource-manager/Microsoft.Compute/2017-10-15-preview/computeArtifactPublishingService.json [hunk 8]
```diff
@@ -0,0 +1,870 @@
...
+          "description": "Properties of the image classic compute profile."
+        },
+        "purchasePlan": {
+          "$ref": "#/definitions/SharedVMImagePurchasePlan",
+          "description": "Properties of the image purchase plan."
+        },
+        "provisioningState": {
+          "readOnly": true,
+          "type": "string"
+        },
```
Comment (olydis) [inline specification/compute/resource-manager/Microsoft.Compute/2017-10-15-preview/computeArtifactPublishingService.json]: Syntax error: trailing comma

File: specification/compute/resource-manager/Microsoft.Compute/2017-10-15-preview/computeArtifactPublishingService.json [hunk 9]
```diff
@@ -0,0 +1,870 @@
...
+          "description": "Properties of the image version publishing profile."
+        },
+		"replicationStatus": {
+          "$ref": "#/definitions/ReplicationStatus",
+          "description": "Properties of the image version replication status."
+        },
+        "provisioningState": {
+          "readOnly": true,
+          "type": "string"
+        },
```
Comment (olydis) [inline specification/compute/resource-manager/Microsoft.Compute/2017-10-15-preview/computeArtifactPublishingService.json]: Syntax error: trailing comma

**Comment by olydis** [comment]
We will close this PR soon due to inactivity. Please address the review comments and resolve the merge conflict.

**Comment by olydis** [comment]
Closing due to inactivity (basic issues not resolved for 2 weeks). Feel free to reopen if you want to move forward.

**Pull request closed without merging**